cfme/services/myservice/ui.py

```diff
@@ -147,115 +148,125 @@ def is_displayed(self):
-
-    def set_ownership(self, owner, group):
-        view = navigate_to(self, 'SetOwnership')
-        view.fill({'select_owner': owner,
-                   'select_group': group})
-        view.save_button.click()
-        view = self.create_view(MyServiceDetailView)
-        assert view.is_displayed
-        view.flash.assert_no_error()
-        view.flash.assert_success_message('Ownership saved for selected Service')
-
-    def edit_tags(self, tag, value):
-        view = navigate_to(self, 'EditTags')
-        view.fill({'select_tag': tag,
-                   'select_value': value})
-        view.save_button.click()
-        view = self.create_view(MyServiceDetailView)
-        assert view.is_displayed
-        view.flash.assert_no_error()
-        view.flash.assert_success_message('Tag edits were successfully saved')
-
-    def check_vm_add(self, add_vm_name):
-        view = navigate_to(self, 'Details')
-        # TODO - replace Quadicon later
-        quadicon = Quadicon(add_vm_name, "vm")
-        sel.click(quadicon)
-        view.flash.assert_no_error()
-
-    def download_file(self, extension):
-        view = navigate_to(self, 'All')
-        view.download_choice.item_select("Download as " + extension)
-        view.flash.assert_no_error()
-
-    def reconfigure_service(self):
-        view = navigate_to(self, 'Reconfigure')
-        view.submit_button.click()
-        # TODO - assert for request view after request widgetastic conversion
-        view.flash.assert_no_error()
+            'Retirement initiated for 1 Service from the {} Database'.format(
+                current_appliance.product_name))
+    # wait for service to retire
+    wait_for(
+        lambda: view.lifecycle_detail.get_text_of("Retirement State") == 'Retiring',
+        fail_func=tb.refresh,
+        num_sec=10 * 60,
+        delay=3,
+        message='Service Retirement wait'
+    )
+
+
+@MyService.retire_on_date.external_implementation_for(ViaUI)
+def retire_on_date(self, retirement_date):
+    view = navigate_to(self, 'SetRetirement')
+    view.retirement_date.fill(retirement_date)
+    view.save_button.click()
+    view = navigate_to(self, 'Details')
+    wait_for(
+        lambda: view.lifecycle_detail.get_text_of("Retirement State") == 'Retiring',
+        fail_func=tb.refresh,
+        num_sec=5 * 60,
+        delay=5,
+        message='Service Retirement'
+    )
+
+
+@MyService.update.external_implementation_for(ViaUI)
+def update(self, updates):
+    view = navigate_to(self, 'Edit')
+    changed = view.fill_with(updates, on_change=view.save_button, no_change=view.cancel_button)
+    view.flash.assert_no_error()
+    if changed:
+        view.flash.assert_success_message('Service "{}" was saved'.format
+            (updates.get('name', self.name)))
+    else:
+        view.flash.assert_success_message('Edit of Service "{}" was cancelled by the \
+            user'.format(updates.get('description', self.description)))
```

Comment:
*The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces.* -- PEP 8

Reply:
Done